python/revng/api/target.py

```diff
@@ -2,38 +2,38 @@
 # This file is distributed under the MIT License. See LICENSE.md for details.
 #
 
+from collections.abc import Sequence
+from itertools import islice
 from typing import Any, Generator, List, Optional
 
 from ._capi import _api, ffi
-from .container import Container
-from .kind import Kind
-from .utils import convert_buffer, make_c_string, make_generator, make_python_string
+from .utils import convert_buffer, make_c_string, make_python_string
 
 
 class Target:
-    def __init__(self, target, container: Container):
+    def __init__(self, target, container):
         self._target = target
         self._container = container
 
     @staticmethod
-    def create(kind: Kind, container: Container, path_components: List[str]) -> Optional["Target"]:
+    def create(_kind, container: str, path_components: List[str]) -> Optional["Target"]:
```

Comment:
As an undocumented convention, variables that start with `_` are used for pointer (e.g. the `_kind` here refers to an `rp_kind *`, not a string or an object)

Reply:
Mmmh, OK, but I think in Python `_` is used to hint that a field is """private""", isn't it? Not sure it's the best convention.

Anyway, we can consider reverting the convention in another moment.

Reply:
> Mmmh, OK, but I think in Python _ is used to hint that a field is """private""", isn't it?

That's (mostly) for classes, also it's more like "please don't touch this unless you're sure". Double underscore invokes mangling which is usually when you want to avoid people touching the attribute

> Not sure it's the best convention.

From PEP8:

> _single_leading_underscore: weak “internal use” indicator

It's a bit of a stretch, but pointers are basically internal to `revng.api`, so it "makes sense" (IMHO)